classes/SpecificPrice.php

```diff
@@ -163,7 +163,9 @@ public static function getIdsByProductId($id_product, $id_product_attribute = fa
 			FROM `'._DB_PREFIX_.'specific_price`
 			WHERE `id_product` = '.(int)$id_product.'
 			AND id_product_attribute='.(int)$id_product_attribute.'
-			AND id_cart='.(int)$id_cart);
+			AND id_cart='.(int)$id_cart.'
+                        AND id_specific_price_rule = 0'
```

Comment:
to be honest I don't understand this condition: can you tell me what do you expect from this new requirement ?